tests/libvirt/fake_libvirt_library.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "tests/fake_handle.h"
+
+#include <libvirt/libvirt.h>
+#include <libvirt/virterror.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+namespace mpt = multipass::test;
+
+/*
+ * These are default fake libvirt functions for testing.  They return the most
+ * common values needed for testing.
+ * See test_libvirt_backend.cpp for examples on how to override these functions
+ * using LibvirtWrapper.
+ */
+
+virConnectPtr virConnectOpen(const char* name)
+{
+    return mpt::fake_handle<virConnectPtr>();
+}
+
+int virConnectClose(virConnectPtr conn)
+{
+    return 0;
+}
+
+char* virConnectGetCapabilities(virConnectPtr /*conn*/)
+{
+    return strdup("");
```

Comment:
The client is supposed to free the allocated string, hence just using `strdup` here.  For reference, https://libvirt.org/html/libvirt-libvirt-host.html#virConnectGetCapabilities

Reply:
You're right, this is fine here. It's on the error msg function that I noticed something wrong and then thought it would be the same in all `char*`-returning functions.